### PR TITLE
Add flow

### PR DESCRIPTION
**Pull request by j-hannah**
@sirkitree Can you take a look at this one? I'd like to confirm that you can get it running without errors. I've updated the project to use yarn so you may have to run `yarn install`. If you run into any snags (or don't have time to do this), let me know.

### CODE REVIEW

File: src/containers/HomePage/HomePage.js [hunk 1]
```diff
@@ -29,13 +31,13 @@ const styles = StyleSheet.create({
 export class HomePage extends PureComponent {
   constructor() {
     super();
-    this.loadMore = this.loadMore.bind(this);
+    (this:any).loadMore = (this:any).loadMore.bind(this);
```
Comment (sirkitree) [inline src/containers/HomePage/HomePage.js]: So we use `(this:any)` instead of `this` everywhere?

File: src/containers/HomePage/HomePage.js [hunk 2]
```diff
@@ -29,13 +31,13 @@ const styles = StyleSheet.create({
...
   }
-  static propTypes = {
-    dispatch: PropTypes.func.isRequired,
-    curatedArticles: PropTypes.arrayOf(PropTypes.object.isRequired),
-    totalCuratedArticles: PropTypes.number,
-    totalCuratedArticlesLoaded: PropTypes.number
+  props: {
+    dispatch: Function,
+    curatedArticles: Array<Object>,
```
Comment (sirkitree) [inline src/containers/HomePage/HomePage.js]: Can you point me to docs or explain `Array<Object>` for me please?

**Comment by sirkitree** [comment]
## Results of my test

### Fresh from `master`
1. `rm -rf node_modules`
1. `npm install -g yarn`
1. `yarn install`
1. noticed yarn is out of date (wtf, I just installed it, ok, fine)
1. `npm upgrade --global yarn`
1. `yarn install`: everything looking good
1. `yarn run server`: localhost:3000 looking good

### Switch into this branch:
1. `git checkout -b add-flow origin/add-flow`
1. `rm -rf node_modules`
1. `yarn install`: everything looking good
1. `yarn run server`: localhost:3000 looking good

File: src/containers/HomePage/HomePage.js (hunk 1 above)
Comment (j-hannah) [inline src/containers/HomePage/HomePage.js]: Only in the case of autobinding in class components. See [here](https://github.com/facebook/flow/issues/1517).

File: src/containers/HomePage/HomePage.js (hunk 2 above)
Comment (j-hannah) [inline src/containers/HomePage/HomePage.js]: Here's the [docs](https://flow.org/en/docs/types/arrays/#toc-array-type) on the Array type annotation and [another reference](https://flow.org/en/docs/frameworks/react/#toc-adding-types-for-react-component-props) on using Flow for typing props in React. You can use both Flow and React's PropTypes together, but for the way I was using it, it didn't add any value.

**Pull request merged**